packages/web3-eth/src/index.ts

```diff
@@ -0,0 +1,371 @@
+			? await RpcMethods.getUncleByBlockHashAndIndex(this._requestManager, block, uncleIndex)
+			: await RpcMethods.getUncleByBlockNumberAndIndex(
+					this._requestManager,
+					block,
+					uncleIndex,
+			  );
+
+		return convertObjectPropertiesToValidType(
+			response,
+			convertibleBlockProperties,
+			returnType ?? this._options.defaultReturnType,
+		);
+	}
+
+	public async getTransaction<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		transactionHash: HexString32Bytes,
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getTransactionByHash(
+			this._requestManager,
+			transactionHash,
+		);
+
+		return response === null
+			? response
+			: convertObjectPropertiesToValidType(
+					response,
+					// TODO
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-expect-error
+					convertibleTransactionInfoProperties,
+					returnType ?? this._options.defaultReturnType,
+			  );
+	}
+
+	// TODO Can't find in spec
+	// public async getPendingTransactions() {
+
+	// }
+
+	public async getTransactionFromBlock<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		block: HexString32Bytes | BlockNumberOrTag = this._options.defaultBlock,
+		transactionIndex: Uint,
+		returnType?: ReturnType,
+	) {
+		const response = isHexString32Bytes(block)
+			? await RpcMethods.getTransactionByBlockHashAndIndex(
+					this._requestManager,
+					block,
+					transactionIndex,
+			  )
+			: await RpcMethods.getTransactionByBlockNumberAndIndex(
+					this._requestManager,
+					block,
+					transactionIndex,
+			  );
+
+		return response === null
+			? response
+			: convertObjectPropertiesToValidType(
+					response,
+					// TODO
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-expect-error
+					convertibleTransactionInfoProperties,
+					returnType ?? this._options.defaultReturnType,
+			  );
+	}
+
+	public async getTransactionReceipt<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		transactionHash: HexString32Bytes,
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getTransactionReceipt(
+			this._requestManager,
+			transactionHash,
+		);
+
+		return response === null
+			? response
+			: convertObjectPropertiesToValidType(
+					response,
+					convertibleReceiptInfoProperties,
+					returnType ?? this._options.defaultReturnType,
+			  );
+	}
+
+	// TODO Discuss the use of multiple optional parameters
+	public async getTransactionCount<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		address: Address,
+		blockNumber: BlockNumberOrTag = this._options.defaultBlock,
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getTransactionCount(
+			this._requestManager,
+			address,
+			blockNumber,
+		);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
+
+	public async sendTransaction(transaction: TransactionWithSender) {
```

Comment:
As discussed earlier the input parameters for this function must match with the `1.x` API. Remember these parameters may differ from the underlaying JSON-RPC API. We may get some extra or some less input and library handle those paremters and process and send to the underlaying transport layer. e.g. The discussion was going around `chainId`, we need to get this input from user and then use it to encode the transaction before sending.

Reply:
Will be addressed in subsequent PR